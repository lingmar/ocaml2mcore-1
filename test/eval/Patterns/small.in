let foo x =
  match x with
    1 -> "one"
  | 2 -> "two"
  | _ when x > 100 -> "big"
  | _ -> "other"

let bar x =
  match x with
    1 -> "bar one"
  | 2 -> "bar two"
  | _ when x > 100 -> "bar big"
  | _ -> "bar other"


let _ =
  print_endline (foo 1);
  print_endline (bar 1);
  print_endline (foo 2);
  print_endline (bar 2);
  print_endline (foo 901);
  print_endline (bar 1989);
  print_endline (foo 0);
  print_endline (bar (-1));
