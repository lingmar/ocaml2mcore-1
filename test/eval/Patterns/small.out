one
bar one
two
bar two
big
bar big
other
bar other
